architecture/additional_concepts/dynamic_admission_controllers.adoc

```diff
@@ -0,0 +1,536 @@
+the reason for doing so is based on the _first_ external admission webhook
+denial reason. 
++
+If more than one admission webhook denies the admission request, 
+only the first will be returned to the user.
+
+* If there is an error encountered when calling an admission
+webhook, that request is ignored and is be used to approve/deny the
+admission request.
+
+The communication between the admission controller and the webhook server needs to be
+secured using  TLS. Generate a CA certificate and use the certificate to sign the server certificate
+used by your webhook. The PEM-formatted CA certificate is supplied
+to the admission controller using a mechanism, such as 
+xref:../../dev_guide/secrets.adoc#service-serving-certificate-secrets[Service Serving Certificate Secrets].
+ 
+[[architecture-additional-concepts-dynamic-admission-webhooks-writing]]
+=== Create the Admission Webhook
+
+First deploy the external webhook server and ensure
+it is working properly. Otherwise, depending whether the webhook is configured as `fail open` or 
+`fail closed`, operations will be unconditionally accepted or rejected. 
+
+. Configure a xref:architecture-additional-concepts-dynamic-admission-webhooks-ex-m[mutating] 
+or xref:architecture-additional-concepts-dynamic-admission-webhooks-ex[validating] admission webhook object in a YAML file.
+
+. Run the following command to create the object:
++
+----
+oc create -f <file-name>.yaml
+----
++
+After you create the admission webhook object, {product-title} takes a few
+seconds to honor the new configuration.
+
+. Create a front-end service for the admission webhook:
++
+[source-yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    role: webhook <1>
+  name: <name>
+spec:
+  selector:
+   role: webhook <1>
+----
++
+<1> Free-form label to trigger the webhook.
+
+. Run the following command to create the object:
++
+----
+oc create -f <file-name>.yaml
+----
+
+. Add the admission webhook name to pods you want controlled by the webhook:
++
+[source-yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    role: webhook <1>
+  name: <name>
+spec:
+  containers:
+    - name: <name>
+      image: myrepo/myimage:latest
+      imagePullPolicy: <policy>
+      ports:
+       - containerPort: 8000
+----
++
+<1> Label to trigger the webhook. 
+
+[NOTE]
+====
+See the link:https://github.com/openshift/kubernetes-namespace-reservation[kubernetes-namespace-reservation projects] 
+for an end-to-end example of how to build your own secure and portable webhook admission server 
+and link:https://github.com/openshift/generic-admission-server[generic-admission-apiserver] for the library.
+====
+
+[[architecture-additional-concepts-dynamic-admission-webhooks-examples]]
+=== Admission Webhook Example
+
+The following is an example admission webhook, named _webhook_, that 
+will not allow pod creation if the webhook server is not available:
```

Comment:
how about "The following is an example admission webhook that will not alllow namespace creation if it's preserved" ?

Reply:
@wanghaoran1988 I am not sure what you mean by preserved. is this a reference to namespace reservation?

Reply:
yes, sorry makes you confusing.